azext_iot/central/providers/device_provider.py

```diff
@@ -240,3 +240,10 @@ def _dps_populate_essential_info(self, dps_info, device_status: DeviceStatus):
             "error": error.get(device_status),
         }
         return filtered_dps_info
+
+    def generate_device_key(
```

Comment:
I'd suggest as a one off not putting this inside a provider, and just calling directly into utils for now

Reply:
or offline, whichever route you go down